command/websocket.go

```diff
@@ -0,0 +1,70 @@
+package command
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io"
+	"time"
+)
+
+// WsURL returns the ws URL for the running browser window.
+// Chrome will sometimes fail to print the websocket, or run for a long
+// time, without properly exiting. To avoid blocking forever in those
+// cases, give up after ten seconds.
+func WsURL(programState *ProgramState) (string, error) {
+	const wsURLReadTimeout = 10 * time.Second
+	var err error
+
+	var wsURL string
+	wsURLChan := make(chan struct{}, 1)
+	go func() {
+		wsURL, err = PollwsURL(programState)
+		wsURLChan <- struct{}{}
+	}()
+	select {
+	case <-wsURLChan:
+	case <-time.After(wsURLReadTimeout):
+		err = errors.New("websocket url timeout reached")
+	}
+
+	if err != nil {
+		return wsURL, err
+	}
+
+	return wsURL, err
+
+}
+
+//PollwsURL retrives the wsURL from the Stderr of the process
+func PollwsURL(programState *ProgramState) (string, error) {
+	prefix := []byte("DevTools listening on")
+	var wsURL string
+	count := 0
+
+	for {
+		out := programState.Stderr()
+		bytesReader := bytes.NewReader(out[count:])
+		bufReader := bufio.NewReader(bytesReader)
+
+		line, err := bufReader.ReadBytes('\n')
+
+		if err == io.EOF {
+			err = nil
+		}
+		if err != nil {
+			return "", err
+		}
+
+		if bytes.HasPrefix(line, prefix) {
```

Comment:
You expect the prefix at the start of your line. When you are trying to use a raw string in test, you have spaces behind your string (probably because you are indenting your string contents). As such, this starts failing in that case.

bytes.HasPrefix(TrimSpace(line), prefix) can work too. Or you can remove the indent in your test main.go.